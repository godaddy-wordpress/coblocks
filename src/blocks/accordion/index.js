/**
 * External dependencies
 */
import { AccordionIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';
import { InnerBlocks } from '@wordpress/block-editor';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	attributes,
	/* translators: block description */
	description: __( 'Organize content within collapsable accordion items.', 'coblocks' ),
	edit,
	example: {
		attributes: {
			count: 2,
		},
	},
	icon: <Icon icon={ icon } />,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'tabs', 'coblocks' ),
		/* translators: block keyword (abbreviation for "frequently asked questions") */
		__( 'faq', 'coblocks' ),
	],
	save() {
		return (
			<div>
				<InnerBlocks.Content />
			</div>
		);
	},
	supports: {
		align: [ 'wide', 'full' ],
		html: false,
	},
	/* translators: block name */
	title: __( 'Accordion', 'coblocks' ),
	transforms,
};

export { name, category, metadata, settings };
