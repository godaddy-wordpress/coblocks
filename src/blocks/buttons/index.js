/**
 * THIS BLOCK IS DEPRECRATED. IT WILL NOT SHOW UP IN THE INSERTER ANYMORE.
 * SEE /src/js/deprecations/deprecate-coblocks-buttons.js
 */

/**
 * Internal dependencies
 */
import deprecated from './deprecated';
import edit from './edit';
import metadata from './block.json';
import save from './save';
import transforms from './transforms';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { button as icon } from '@wordpress/icons';

/**
 * Block constants
 */
const { name, category, attributes } = metadata;

const settings = {
	/* translators: block name */
	title: __( 'Buttons', 'coblocks' ),
	/* translators: block description */
	description: __( 'Prompt visitors to take action with multiple buttons, side by side.', 'coblocks' ),
	icon,
	keywords: [
		'coblocks',
		/* translators: block keyword */
		__( 'link', 'coblocks' ),
		/* translators: block keyword (abbreviation for "call to action") */
		__( 'cta', 'coblocks' ),
		/* translators: block keyword */
		__( 'call to action', 'coblocks' ),
	],
	attributes,
	transforms,
	edit,
	save,
	deprecated,
};

export { name, category, metadata, settings };
