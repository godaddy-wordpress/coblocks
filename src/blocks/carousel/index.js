/**
 * WordPress dependencies
 */
import { InnerBlocks, useBlockProps } from '@wordpress/block-editor';

/**
 * Internal dependencies
 */
import BlockEdit from './block-edit';
import metadata from './block.json';

const settings = {
	edit: BlockEdit,
	save: () => (
		<div { ...useBlockProps.save() }>
			<InnerBlocks.Content />
		</div>
	),
};

const name = metadata.name;
export { name, metadata, settings };
