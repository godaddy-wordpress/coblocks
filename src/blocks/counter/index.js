/**
 * External dependencies
 */
import { CounterIcon as icon } from '@godaddy-wordpress/coblocks-icons';

/**
 * WordPress dependencies
 */
import { __ } from '@wordpress/i18n';
import { Icon } from '@wordpress/components';

/**
 * Internal dependencies
 */
import edit from './edit';
import metadata from './block.json';
import save from './save';

/**
 * Block constants
 */
const { name, category } = metadata;

const attributes = {
	...metadata.attributes,
};

const settings = {
	attributes,
	edit,
	example: {
		attributes: {
			align: 'center',
			counterDescription: ' ',
			counterText: '10 years, 2 months, 20 days, 4 hours',
			fontSize: 'medium',
		},
	},
	icon: <Icon icon={ icon } />,
	save,
	title: __( 'Counter', 'coblocks' ),
};

export { name, category, metadata, settings };

