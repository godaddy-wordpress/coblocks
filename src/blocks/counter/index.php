<?php
/**
 * This file defines a function and adds an action to register Counter patterns.
 *
 * @package CoBlocks
 */

/**
 * Registers the Counter block patterns.
 */
function coblocks_counter_register_block_patterns() {
	// We include viewport when version is below 5.9.
	// WordPress 5.9 renders pattern previews differently.
	$should_include_viewport = ! is_wp_version_compatible( '5.9' );

	$icon_array = array(
		'title'      => __( 'Counter and Icon block Group', 'coblocks' ),
		'blockTypes' => array( 'core/group', 'coblocks/counter', 'coblocks/icon' ),
		'content'    => '<!-- wp:group {"backgroundColor":"tertiary"} --><div class="wp-block-group has-tertiary-background-color has-background"><!-- wp:coblocks/icon {"icon":"build","contentAlign":"center","iconColor":"primary","className":"is-style-outlined"} /--><!-- wp:coblocks/counter {"align":"center","counterDescription":"Hours Worked","counterText":"\u003cstrong\u003e1000\u003c/strong\u003e","textColor":"primary","fontSize":"small"} --><div class="wp-block-coblocks-counter has-text-align-center has-primary-color has-text-color has-small-font-size" data-counter-speed="1"><strong>1000</strong></div><div class="wp-block-coblocks-counter has-text-align-center has-primary-color has-text-color has-small-font-size" data-counter-speed="1">Hours Worked</div><!-- /wp:coblocks/counter --></div><!-- /wp:group -->',
	);

	$multi_digits_array = array(
		'title'      => __( 'Counter with multiple inline digits', 'coblocks' ),
		'blockTypes' => array( 'coblocks/counter' ),
		'content'    => '<!-- wp:coblocks/counter {"align":"center","counterDescription":"Since Last Incident","counterText":"\u003cmeta charset=\u0022utf-8\u0022\u003e10 days, 20 hours","fontSize":"small"} --><div class="wp-block-coblocks-counter has-text-align-center has-small-font-size" data-counter-speed="1"><meta charset="utf-8">10 days, 20 hours</div><div class="wp-block-coblocks-counter has-text-align-center has-small-font-size" data-counter-speed="1">Since Last Incident</div><!-- /wp:coblocks/counter -->',
	);

	if ( $should_include_viewport ) {
		foreach ( array( $icon_array, $multi_digits_array ) as &$pattern_array ) {
			array_push( $pattern_array, array( 'viewport' => '150' ) );
		};
		unset( $pattern_array );
	}

	register_block_pattern( 'counter/with-icon', $icon_array );
	register_block_pattern( 'counter/with-multiple-digits', $multi_digits_array );

}

add_action( 'init', 'coblocks_counter_register_block_patterns' );

