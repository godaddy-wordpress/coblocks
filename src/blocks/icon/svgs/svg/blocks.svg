<svg fill="none" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path d="m10.3519669 11.7141378v8.2858622h8.2815735v-8.2858622zm-10.3519669 8.2858622h8.2815735v-8.2858622h-8.2815735zm0-18.64319006v8.28586225h8.2815735v-8.28586225zm14.1407867-1.35680994-5.8592132 5.85189021 5.8592132 5.86224759 5.8592133-5.86224759z" fill="currentColor" /></svg>