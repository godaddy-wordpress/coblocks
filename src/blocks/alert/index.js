/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const { replaceBlocks } = dispatch( 'core/block-editor' );

		replaceBlocks(
			[ props.clientId ],
			createBlock( 'core/paragraph', props.attributes ),
		);
		return null;
	},
	parent: [],
	save: () => null,
	title: metadata.title,
};

export { name, metadata, settings };
