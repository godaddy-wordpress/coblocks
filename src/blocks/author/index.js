/**
 * Internal dependencies
 */
import metadata from './block.json';

/**
 * WordPress dependencies
 */
import { createBlock } from '@wordpress/blocks';
import { dispatch, select } from '@wordpress/data';

/**
 * Block constants
 */
const { name } = metadata;

const settings = {
	edit: ( props ) => {
		const parentBlock = select( 'core/editor' ).getBlocksByClientId( props.clientId )[ 0 ];
		if ( ! parentBlock ) {
			return null;
		}

		// Used to set text color props
		const hasTextColor = ( props.attributes.hasOwnProperty( 'textColor' ) );
		const hasCustomTextColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'text' ) );

		// Used to set background color props
		const hasBackgroundColor = ( props.attributes.hasOwnProperty( 'backgroundColor' ) );
		const hasCustomBackgroundColor = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'color' ) && props.attributes.style.color.hasOwnProperty( 'background' ) );

		// Used to set font props
		const hasFontSize = ( props.attributes.hasOwnProperty( 'fontSize' ) );
		const hasCustomFont = ( props.attributes.hasOwnProperty( 'style' ) &&
			props.attributes.style.hasOwnProperty( 'typography' ) );

		const { replaceBlocks } = dispatch( 'core/block-editor' );
		const isRTL = select( 'core/editor' ).getEditorSettings().isRTL;

		const authorNameBlock = () => {
			let authorNameBlockProps = {
				content: props.attributes.name,
				level: 4,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			// Headings do not inherit color from parent - we descend the style here for initial match.
			if ( hasCustomTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					style: { ...authorNameBlockProps?.style,
						color: { ...authorNameBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				authorNameBlockProps = { ...authorNameBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			return createBlock( 'core/paragraph', authorNameBlockProps );
		};

		const authorBioBlock = () => {
			let authorBioBlockProps = {
				content: props.attributes.biography,
			};

			// Font controls - default is fontSize and style is used for custom setting.
			// All font controls should apply to the bio and name.
			if ( hasFontSize ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					fontSize: props.attributes.fontSize,
				};
			}
			if ( hasCustomFont ) {
				authorBioBlockProps = {
					...authorBioBlockProps,
					style: {
						...authorBioBlockProps?.style,
						typography: props.attributes.style.typography,
					},
				};
			}

			return createBlock( 'core/paragraph', authorBioBlockProps );
		};

		const buttonBlock = parentBlock.innerBlocks;
		// Set buttons to align.
		buttonBlock.forEach( function( button, index ) {
			buttonBlock[ index ].attributes.align = isRTL ? 'right' : 'left';
		} );
		// Button blocks are no-longer stand-alone and should be within `core/buttons` block.
		const coreButtons = createBlock( 'core/buttons', { }, [ ...buttonBlock ] );

		const imageBlock = createBlock( 'core/image', { className: 'is-style-rounded', id: props.attributes.imgId, url: props.attributes.imgUrl } );

		const leftColumn = createBlock( 'core/column', { width: '25%' }, [ imageBlock ] );
		const rightColumn = createBlock( 'core/column', { width: '75%' }, [ authorNameBlock(), authorBioBlock(), coreButtons ] );

		const columnsBlock = () => {
			let columnsBlockProps = {
				...( props.attributes.hasOwnProperty( 'className' ) && { className: props.attributes.className } ),
				...( props.attributes.hasOwnProperty( 'animation' ) && { animation: props.attributes.animation } ),
				...( props.attributes.hasOwnProperty( 'anchor' ) && { anchor: props.attributes.anchor } ),
				style: {
					spacing: {
						padding: {
							bottom: '2.5rem',
							left: '2.5rem',
							right: '2.5rem',
							top: '2.5rem',
						},
					},
				},
			};

			// Migrate text colors into columns block props.
			if ( hasCustomTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							text: props.attributes.style.color.text,
						},
					},
				};
			}
			if ( hasTextColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					textColor: props.attributes.textColor,
				};
			}

			// Migrate background colors into columns block props.
			if ( hasCustomBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: props.attributes.style.color.background,
						},
					},
				};
			}
			if ( hasBackgroundColor ) {
				columnsBlockProps = { ...columnsBlockProps,
					backgroundColor: props.attributes.backgroundColor,
				};
			}

			// Does not have background color set. Use default block color from CoBlocks.
			const hasNoBackgroundColorSet = ! props.attributes.hasOwnProperty( 'backgroundColor' ) &&
				! props.attributes?.style?.color?.hasOwnProperty( 'background' );
			if ( hasNoBackgroundColorSet ) {
				// $dark-opacity-light-200 computed #8C8C971A - no variable access in root.
				columnsBlockProps = { ...columnsBlockProps,
					style: { ...columnsBlockProps?.style,
						color: { ...columnsBlockProps?.style?.color,
							background: '#8C8C971A',
						},
					},
				};
			}

			return createBlock( 'core/columns', columnsBlockProps, [ leftColumn, rightColumn ] );
		};

		replaceBlocks(
			[ props.clientId ],
			columnsBlock()
		);

		return null;
	},
	parent: [],
	save: () => null,
	/* translators: block name */
	title: metadata.title,
};

export { name, metadata, settings };
