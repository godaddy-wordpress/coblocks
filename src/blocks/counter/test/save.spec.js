/**
 * WordPress dependencies
 */
 import { registerCoreBlocks } from '@wordpress/block-library';
 import { registerBlockType, createBlock, serialize } from '@wordpress/blocks';

registerCoreBlocks();

/**
 * External dependencies
 */
import '@testing-library/jest-dom/extend-expect';

/**
 * Internal dependencies.
 */
import { name, settings } from '../index';

// Make variables accessible for all tests.
let block;
let serializedBlock;

const baseAttributes = {
	counterDescription: 'Always always tested 1000 times',
	counterText: '10000 hours 20000 days',
};

// Counter block uses no innerBlocks.
const baseInnerBlocks = [ ]

describe( name, () => {
	beforeAll( () => {
		// Register the block.
		registerBlockType( name, { category: 'common', ...settings } );
	} );

	beforeEach( () => {
		// Create the block with the minimum attributes.
		block = createBlock( name, baseAttributes, baseInnerBlocks );
		// Reset the reused variables.
		serializedBlock = '';
	} );

	afterEach( () => {
		// Make a snapshot for each save function test to better detect deprecation needs.
		expect( serializedBlock ).toMatchSnapshot();
	} );

	it( 'should have data attribute \'[data-counter-speed]\' with counterSpeed value.', () => {
		block.attributes = { ...block.attributes, counterSpeed: 5 };
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'data-counter-speed=\"5\"' );
	} );

	it( 'should have attribute \'counterText\' defined.', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'counterText' );
	} );

	it( 'should have attribute \'counterDescription\' defined.', () => {
		serializedBlock = serialize( block );

		expect( serializedBlock ).toBeDefined();
		expect( serializedBlock ).toContain( 'counterDescription' );
	} );

    [ 'left', 'center', 'right' ].forEach( ( align ) => {
		it( `should have className 'has-text-align-${ align }' with align set to '${align}'.`, () => {
			block.attributes = { ...block.attributes, align };

			serializedBlock = serialize( block );

			expect( serializedBlock ).toBeDefined();
			expect( serializedBlock ).toContain( `has-text-align-${ align }` );
		} );
	} );
} );
