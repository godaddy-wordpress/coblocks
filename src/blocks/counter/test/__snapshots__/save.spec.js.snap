// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coblocks/counter should have attribute 'counterDescription' defined. 1`] = `
"<!-- wp:coblocks/counter {\\"counterDescription\\":\\"Always always tested 1000 times\\",\\"counterText\\":\\"10000 hours 20000 days\\"} -->
<div class=\\"wp-block-coblocks-counter\\" data-counter-speed=\\"1\\">10000 hours 20000 days</div><div class=\\"wp-block-coblocks-counter\\" data-counter-speed=\\"1\\">Always always tested 1000 times</div>
<!-- /wp:coblocks/counter -->"
`;

exports[`coblocks/counter should have attribute 'counterText' defined. 1`] = `
"<!-- wp:coblocks/counter {\\"counterDescription\\":\\"Always always tested 1000 times\\",\\"counterText\\":\\"10000 hours 20000 days\\"} -->
<div class=\\"wp-block-coblocks-counter\\" data-counter-speed=\\"1\\">10000 hours 20000 days</div><div class=\\"wp-block-coblocks-counter\\" data-counter-speed=\\"1\\">Always always tested 1000 times</div>
<!-- /wp:coblocks/counter -->"
`;

exports[`coblocks/counter should have className 'has-text-align-center' with align set to 'center'. 1`] = `
"<!-- wp:coblocks/counter {\\"align\\":\\"center\\",\\"counterDescription\\":\\"Always always tested 1000 times\\",\\"counterText\\":\\"10000 hours 20000 days\\"} -->
<div class=\\"wp-block-coblocks-counter has-text-align-center\\" data-counter-speed=\\"1\\">10000 hours 20000 days</div><div class=\\"wp-block-coblocks-counter has-text-align-center\\" data-counter-speed=\\"1\\">Always always tested 1000 times</div>
<!-- /wp:coblocks/counter -->"
`;

exports[`coblocks/counter should have className 'has-text-align-left' with align set to 'left'. 1`] = `
"<!-- wp:coblocks/counter {\\"align\\":\\"left\\",\\"counterDescription\\":\\"Always always tested 1000 times\\",\\"counterText\\":\\"10000 hours 20000 days\\"} -->
<div class=\\"wp-block-coblocks-counter has-text-align-left\\" data-counter-speed=\\"1\\">10000 hours 20000 days</div><div class=\\"wp-block-coblocks-counter has-text-align-left\\" data-counter-speed=\\"1\\">Always always tested 1000 times</div>
<!-- /wp:coblocks/counter -->"
`;

exports[`coblocks/counter should have className 'has-text-align-right' with align set to 'right'. 1`] = `
"<!-- wp:coblocks/counter {\\"align\\":\\"right\\",\\"counterDescription\\":\\"Always always tested 1000 times\\",\\"counterText\\":\\"10000 hours 20000 days\\"} -->
<div class=\\"wp-block-coblocks-counter has-text-align-right\\" data-counter-speed=\\"1\\">10000 hours 20000 days</div><div class=\\"wp-block-coblocks-counter has-text-align-right\\" data-counter-speed=\\"1\\">Always always tested 1000 times</div>
<!-- /wp:coblocks/counter -->"
`;

exports[`coblocks/counter should have data attribute '[data-counter-speed]' with counterSpeed value. 1`] = `
"<!-- wp:coblocks/counter {\\"counterDescription\\":\\"Always always tested 1000 times\\",\\"counterSpeed\\":5,\\"counterText\\":\\"10000 hours 20000 days\\"} -->
<div class=\\"wp-block-coblocks-counter\\" data-counter-speed=\\"5\\">10000 hours 20000 days</div><div class=\\"wp-block-coblocks-counter\\" data-counter-speed=\\"5\\">Always always tested 1000 times</div>
<!-- /wp:coblocks/counter -->"
`;
